prometheus_test.go

```diff
@@ -448,3 +449,88 @@ tests_foo{key_1="issue659",key_2="",key_3="issue659",key_4="",key_5="issue659"}
 		t.Fatalf("output differed from expected output: %s want: %s", output, want)
 	}
 }
+
+func TestShareDefaultRegistry(t *testing.T) {
+	_, err := NewExporter(Options{
+		Registerer: prometheus.DefaultRegisterer,
+		Gatherer:   prometheus.DefaultGatherer,
+	})
+	if err != nil {
+		t.Fatalf("failed to create prometheus exporter: %v", err)
+	}
+	m := stats.Int64("tests/foo", "foo", stats.UnitDimensionless)
+	v := &view.View{
+		Name:        m.Name(),
+		Description: m.Description(),
+		Measure:     m,
+		Aggregation: view.Count(),
+	}
+	if err := view.Register(v); err != nil {
+		t.Fatalf("failed to create views: %v", err)
+	}
+	defer view.Unregister(v)
+	view.SetReportingPeriod(time.Millisecond)
+
+	stats.Record(context.Background(), m.M(1))
+
+	// counter, prometheus way
+	c := prometheus.NewCounter(prometheus.CounterOpts{Name: "prom_counter", Help: "Prometheus Counter"})
+	prometheus.MustRegister(c)
+
+	c.Add(1)
+
+	// Use prometheus handler
+	srv := httptest.NewServer(promhttp.Handler())
+	defer srv.Close()
+
+	var i int
+	var output string
+	for {
+		time.Sleep(10 * time.Millisecond)
+		if i == 1000 {
```

Comment:
nit: may be 10 is sufficient to declare it failed. Oc counter should be available right away. Not sure about Prometheus counter though.

Reply:
Thank you for your review. All updates are done, please take a look.